Box.V2/Managers/BoxTermsOfServiceManager.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Box.V2.Auth;
+using Box.V2.Config;
+using Box.V2.Converter;
+using Box.V2.Exceptions;
+using Box.V2.Extensions;
+using Box.V2.Models;
+using Box.V2.Services;
+using Box.V2.Utility;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+
+
+
+namespace Box.V2.Managers
+{
+    /// <summary>
+    /// The manager that represents all of the terms of service endpoints
+    /// </summary>
+    public class BoxTermsOfServicesManager : BoxResourceManager
+    {
+        public BoxTermsOfServicesManager(IBoxConfig config, IBoxService service, IBoxConverter converter, IAuthRepository auth, string asUser = null, bool? suppressNotifications = null)
+            : base(config, service, converter, auth, asUser, suppressNotifications) { }
+
+
+        /// <summary>
+        /// Retrieves information about a terms of service by ID
+        /// </summary>
+        /// <param name="tosId">Id of the terms of service.</param>
+        /// <returns>A full terms of service object is returned if the id is valid and if the user has access to the terms of service.</returns>
+        public async Task<BoxTermsOfService> GetTermsOfServicesByIdAsync(string tosId)
+        {
+            tosId.ThrowIfNullOrWhiteSpace("tosId");
+
+            BoxRequest request = new BoxRequest(_config.TermsOfServicesEndpointUri, tosId);
+
+            IBoxResponse<BoxTermsOfService> response = await ToResponseAsync<BoxTermsOfService>(request).ConfigureAwait(false);
+
+            return response.ResponseObject;
+        }
+
+        /// <summary>
+        /// Retrieves information about all terms of service
+        /// </summary>
+        /// <param name="tosType">The type of terms of service to be retrieved - managed or external.</param>
+        /// <returns>A a collection of terms of service objects is returned if the id if the user has access.</returns>
+        public async Task<BoxTermsOfServiceCollection<BoxTermsOfService>> GetTermsOfServicesAsync(string tosType = null)
```

Comment:
what's the limit on this api?
do we support paging?
what if 10000+ tos returned?